src/main/groovy/life/qbic/datamodel/accounting/Offer.groovy

```diff
@@ -59,24 +61,67 @@ class Offer {
      */
     final OfferId identifier
 
-    Offer(Date modificationDate, Date expirationDate, Person customer, Person projectManager, String projectTitle, String projectDescription, List<ProductItem> items, double totalPrice, OfferId identifier) {
-        this.modificationDate = modificationDate
-        this.expirationDate = expirationDate
-        this.customer = customer
-        this.projectManager = projectManager
-        this.projectTitle = projectTitle
-        this.projectDescription = projectDescription
-        this.items = items
-        this.totalPrice = totalPrice
-        this.identifier = identifier
+    static class Builder {
+
+        Date modificationDate
+
+        Date expirationDate
+
+        Customer customer
+
+        Person projectManager
+
+        String projectTitle
+
+        String projectDescription
+
+        List<ProductItem> items
+
+        double totalPrice
+
+        OfferId identifier
+
+        Builder(Date modificationDate, Date expirationDate, Customer customer, Person projectManager, String projectDescription, String projectTitle, List<ProductItem> items, double totalPrice, OfferId identifier) {
+            this.modificationDate = modificationDate
+            this.expirationDate = expirationDate
+            this.customer = customer
+            this.projectManager = projectManager
+            this.projectDescription = projectDescription
+            this.projectTitle = projectTitle
+            this.items = items
+            this.totalPrice = totalPrice
+            this.identifier = identifier
+        }
```

Comment:
I am missing a default constructor. Otherwise, this builder does not make sense because it only can be instantiated with all the information. There is no flexibility gained with this approach.

There should be a constructor with minimal information and default values for all fields.